### PR TITLE
Situations data coverage choropleth tweaks

### PR DESCRIPTION
**Pull request by VWSCoronaDashboard1**
## Summary

- Fix centering of the icon within the bubble for the legend items
- Prevent hidden button from receiving first focus when opening the choropleth in fullscreen mode

## Motivation

Design request.

## Detailed design

In the case of this choropleth the first button in the modal turned out to be the hidden 'tab navigation' button. Default behavior of the `useFocusTrap` is apparently to give focus on the first focusable item it can find. In previous cases this was a visible button, only here it was setting focus, and thereby revealing, the 'tab navigation' button.
So, I've added an ID to the close button and pass this ID on to the focusSelector option of the `useFocusTrap` hook, that way the close button always receives first focus.

### Governance

- [ ] Documentation is added
- [ ] Test cases are added or updated
- [ ] I've read the contributing document https://github.com/minvws/.github/blob/master/CONTRIBUTING.md
- [ ] I've read and understand the Code of Conduct https://github.com/minvws/.github/blob/master/CODE_OF_CONDUCT.md
- [ ] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License
      https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/LICENSE.txt
      and the contributor license agreement https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/CLA.md

### CODE REVIEW

**Review by VWSCoronaDashboard8 — APPROVED** [review]

File: packages/app/src/components/fullscreen-chart-tile.tsx [hunk 1]
```diff
@@ -65,6 +68,7 @@ export function FullscreenChartTile({
             title={label}
             onClick={() => setIsFullscreen((x) => !x)}
             size={36}
+            id={isFullscreen ? closeButtonId : undefined}
```
Comment (VWSCoronaDashboard10) [inline packages/app/src/components/fullscreen-chart-tile.tsx:71]: I think these elements (the `tile` variable) get rendered twice in the DOM meaning there are duplicate id's.

**Review by VWSCoronaDashboard10 — CHANGES_REQUESTED** [review]
I think the focus thing can be implemented a little bit cleaner:

```diff
diff --git a/packages/app/src/components/modal.tsx b/packages/app/src/components/modal.tsx
index 7b531621..213b2a28 100644
--- a/packages/app/src/components/modal.tsx
+++ b/packages/app/src/components/modal.tsx
@@ -15,7 +15,7 @@ interface ModalProps {
 export function Modal({ id, children, onClose, isFullheight }: ModalProps) {
   const clickRef = useRef<HTMLDivElement>(null);
   useHotkey('esc', onClose);
-  const focusRef = useFocusTrap();
+  const focusRef = useFocusTrap(true, { focusSelector: `#${id}` });
 
   return (
     <ModalPortal id={id}>
@@ -25,6 +25,7 @@ export function Modal({ id, children, onClose, isFullheight }: ModalProps) {
           css={css({ p: 5, height: isFullheight ? '100%' : undefined })}
           onClick={(evt) => evt.target === clickRef.current && onClose()}
           ref={clickRef}
+          id={id}
         >
           {children}
         </div>
```

This tells the focus trap to set focus on the containing div (inside the modal), which in fact cannot receive focus so there's no visible focus. But a single tab-press will move the focus to the first element.

If instead it is a requirement to have the initial focus on the close button (can't imagine that, but who knows), then you could also move the close button to be the first child of the modal content instead of being the last.

File: packages/app/src/components/fullscreen-chart-tile.tsx (hunk 1 above)
Comment (VWSCoronaDashboard1) [reply packages/app/src/components/fullscreen-chart-tile.tsx]: Yup, and setting the focus on the containing div of the modal worked too. Cheers!

File: packages/app/src/components/modal.tsx [hunk 2]
```diff
@@ -4,26 +4,21 @@ import { ReactNode, useEffect, useRef, useState } from 'react';
...
-export function Modal({
-  id,
-  children,
-  onClose,
-  isFullheight,
-  focusSelector,
-}: ModalProps) {
+export function Modal({ id, children, onClose, isFullheight }: ModalProps) {
   const clickRef = useRef<HTMLDivElement>(null);
+  const divId = useUniqueId();
```
Comment (VWSCoronaDashboard10) [inline packages/app/src/components/modal.tsx:18]: nit;
```suggestion
  const focusId = useUniqueId();
```

**Review by VWSCoronaDashboard10 — APPROVED** [review]

**Pull request merged**